### PR TITLE
Internal zones fix

### PR DESCRIPTION
**Pull request by ViktoriiaKovalova**

### CODE REVIEW

File: pkg/automation/recommendations.go [hunk 1]
```diff
@@ -43,12 +49,27 @@ func (s *googleService) ListRecommendations(project, location, recommenderID str
...
-	return recommendations, nil
+	err := DoRequestWithRetries(func() error {
+		recommendations = nil
+		err := listCall.Pages(s.ctx, addRecommendations)
+		// Check if error is because current location is not available for getting recommendations.
+		if googleErr, ok := err.(*googleapi.Error); ok {
+			body := googleErr.Body
+			var fields map[string]interface{}
+			if parseErr := json.Unmarshal([]byte(body), &fields); parseErr != nil {
+				return err
```
Comment (minorczyka) [inline pkg/automation/recommendations.go:60]: Let's not return this error, it's better to return the original error.

File: pkg/automation/recommendations.go [hunk 2]
```diff
@@ -43,12 +49,27 @@ func (s *googleService) ListRecommendations(project, location, recommenderID str
...
+			}
+			if errorJSON, ok := fields["error"].(map[string]interface{}); ok {
+				if status, ok := errorJSON["status"].(string); ok && status == "INVALID_ARGUMENT" {
+					log.Printf("Invalid location error: %v received while getting recommendations for %s %s %s",
+						body, project, location, recommenderID)
+					return nil
+				}
+			}
```
Comment (minorczyka) [inline pkg/automation/recommendations.go:68]: Can we move this check to separate function? It would make `ListRecommendations` function more readable.

**Review by minorczyka — CHANGES_REQUESTED** [review]
I can see changes from the second PR.

File: pkg/automation/recommendations.go (hunk 1 above)
Comment (ViktoriiaKovalova) [reply pkg/automation/recommendations.go:60]: it's the original error. I made a check with return here to avoid having many nested ifs.

**Review by minorczyka — APPROVED** [review]

**Pull request merged**